src/main/scala/chisel3/util/Valid.scala

```diff
@@ -173,6 +178,15 @@ object Pipe {
   */
 class Pipe[T <: Data](val gen: T, val latency: Int = 1) extends Module {
 
+  /** A non-ambiguous name of this `Pipe` for use in generated Verilog names.
+   * Includes the cycle count in the name as well as the parameterized generator's
+   * `typeName`, e.g. `Pipe_4cycles_UInt4`
+   */
+  override def desiredName = {
+    val latencyStr = s"${latency}cycle${if (latency > 1) "s" else ""}"
+    s"${this.getClass.getSimpleName}_${latencyStr}_${gen.typeName}"
```

Comment:
why does this one have an underscore before latency, whereas queue depth does not have an underscore? In general what is the suggested pattern for naming things?

Reply:
could even hash the parameters or something